crates/autopilot/src/infra/solvers/authenticator.rs

```diff
@@ -0,0 +1,86 @@
+use {
+    crate::{
+        domain,
+        infra::blockchain::{contracts::deployment_address, ChainId},
+    },
+    ethcontract::{dyns::DynWeb3, transaction::TransactionResult},
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Authenticator {
+    /// The authenticator contract used for allow-listing solvers to settle.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to add/remove solvers.
+    authenticator_eoa: ethcontract::Account,
+}
+
+impl Authenticator {
+    /// Creates an authenticator which can remove solvers from the allow-list
+    pub async fn new(
+        web3: DynWeb3,
+        chain: ChainId,
+        settlement: contracts::GPv2Settlement,
+        authenticator_eoa: ethcontract::Account,
```

Comment:
nit: I would maybe consider constructor to receive 

```
pub struct Addresses {
    pub settlement: eth::Address,
    pub authenticator_eoa: eth::H256,
}
```

instead of already prepared `settlement` and `authenticator_eoa` objects. This would make it a bit easier to extract this file out of autopilot if needed - we want to have more dependencies on the raw configuration data.

Reply:
@sunce86 but if I do that, then I need to rebuild this:

```rust
        let settlement = contracts::GPv2Settlement::at(
            web3,
            address_for(
                contracts::GPv2Settlement::raw_contract(),
                addresses.settlement,
            ),
        );
```